### PR TITLE
feat: submit & leaderboard routes

### PR DESCRIPTION
**Pull request by vidhik2002**

### CODE REVIEW

File: app.js [hunk 1]
```diff
@@ -0,0 +1,60 @@
+const express = require("express")
+const app = express();
+const mongoose = require("mongoose") 
+const bp = require("body-parser")
+require('dotenv/config')
+// const User = require("../models/User");
+// const Question = require("../models/Question");
+// const Map = require("../models/Map");
+
+app.use(bp.urlencoded({extended: false}))
```
Comment (alias-rahil) [inline app.js]: use [express's built-in body parser](http://expressjs.com/en/api.html). You can use it like, `express.json()` or `express.urlencoded({extended: false})`.

File: package.json [hunk 2]
```diff
@@ -0,0 +1,20 @@
+{
+  "name": "hii",
```
Comment (alias-rahil) [inline package.json]: Change `hii` to `riddler-backend` maybe?

Comment (roerohan) [inline package.json]: Package name fix pls

File: routes/leaderboard.js [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+    username,
+    score,
+    }));
+    res.status(200).json(responseJSON);
+} catch (e) {
+    console.log(e);
+    res.status(500).json({
+    error: e,
+    });
+}
```
Comment (roerohan) [inline routes/leaderboard.js]: Indentation

File: package.json [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+    "start": "nodemon app.js"
+  },
+  "author": "vidhi",
+  "license": "ISC",
+  "description": "",
+  "dependencies": {
+    "body-parser": "^1.19.0",
+    "dotenv": "^8.2.0",
+    "express": "^4.17.1",
+    "fs": "^0.0.1-security",
```
Comment (alias-rahil) [inline package.json]: remove `fs`

File: app.js [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+
+app.get("/", (req, res) => {
+  res.send("homepage");
+});
+
+mongoose.connect(
+    process.env.DB_CONNECTION,
+    {useNewUrlParser: true,
+	  useCreateIndex: true,
+    useUnifiedTopology: true,}
```
Comment (roerohan) [inline app.js]: Use linter

**Review by alias-rahil — CHANGES_REQUESTED** [review]

File: app.js [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+const bp = require("body-parser")
+require('dotenv/config')
+// const User = require("../models/User");
+// const Question = require("../models/Question");
+// const Map = require("../models/Map");
+
+app.use(bp.urlencoded({extended: false}))
+app.use(bp.json())
+
+const testRoute = require('./routes/testroute')
```
Comment (roerohan) [inline app.js]: All requires are supposed to be on the start of the file

**Pull request closed without merging**